src/Analyzers/Core/Analyzers/Helpers/DiagnosticHelper.cs

```diff
@@ -84,11 +84,45 @@ internal static class DiagnosticHelper
             IEnumerable<Location> additionalLocations,
             IDictionary<string, IEnumerable<int>> tagIndices,
             params object[] messageArgs)
+        {
+            return CreateWithLocationTags(descriptor, location, effectiveSeverity, additionalLocations, tagIndices, ImmutableDictionary<string, string>.Empty, messageArgs);
+        }
+
+        /// <summary>
+        /// Create a diagnostic that adds properties specifying a tag for a set of locations.
+        /// </summary>
+        /// <param name="descriptor">A <see cref="DiagnosticDescriptor"/> describing the diagnostic.</param>
+        /// <param name="location">An optional primary location of the diagnostic. If null, <see cref="Location"/> will return <see cref="Location.None"/>.</param>
```

Comment:
ick.  do we have to allow null?  who passes that in?

Reply:
No idea. Copied from existing method so I could add properties.